cli/src/alluxio.org/cli/processes/common.go

```diff
@@ -184,35 +173,37 @@ func addStartFlags(argument string, cmd *env.StartProcessCommand) string {
 	return strings.Join(command, " ")
 }
 
-func addStopFlags(argument string, cmd *env.StopProcessCommand) string {
-	cliPath := filepath.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), "bin", "alluxio")
+func addStopFlags(cmd *env.StopProcessCommand, arguments ...string) string {
+	cliPath := filepath.Join(env.Env.EnvVar.GetString(env.ConfAlluxioHome.EnvVar), names.BinAlluxio)
 	var command []string
-	command = append(command, cliPath, argument)
+	command = append(command, cliPath)
+	command = append(command, arguments...)
 	if cmd.SoftKill {
 		command = append(command, "-s")
 	}
 	return strings.Join(command, " ")
 }
 
-func getHostnames(mode string) ([]string, error) {
-	if mode != "master" && mode != "worker" && mode != "all" {
-		return nil, stacktrace.NewError("invalid mode for readHostnames, " +
-			"available readHostnames modes: [master, worker, all]")
-	}
+const (
+	HostGroupMasters = "masters"
+	HostGroupWorkers = "workers"
+)
+
+func getHostnames(hostGroups []string) ([]string, error) {
 	var hosts []string
-	if mode == "master" || mode == "all" {
-		masterList, err := NewHostnamesFile("masters").getHostnames()
-		if err != nil {
-			return nil, stacktrace.Propagate(err, "cannot get masters")
-		}
-		hosts = append(hosts, masterList...)
-	}
-	if mode == "worker" || mode == "all" {
-		workerList, err := NewHostnamesFile("worker").getHostnames()
-		if err != nil {
-			return nil, stacktrace.Propagate(err, "cannot get workers")
+	for _, hostGroup := range hostGroups {
+		switch hostGroup {
+		case HostGroupMasters:
+		case HostGroupWorkers:
```

Comment:
If the host group is 'masters' this part will do nothing. As a consequence, no masters can be found.